### PR TITLE
Add exclusive list and select list

### PR DESCRIPTION
**Pull request by jsmccrumb**
I implemented a version of select and exclusive lists (based on description in the issues I opened). Works fine on my end but I don't know your code too well and since its not a particularly minor change, it is possible that I broke something in the process.

### CODE REVIEW

File: js/structured-filter.js [hunk 1]
```diff
@@ -153,7 +155,7 @@ $.widget( 'evol.structFilter', {
 				that._field=that._getFieldById(fieldID);
 				var fType=that._type=that._field.type;
 				that._setEditorOperator();
-				if(fType==fTypes.list || fType==fTypes.bool){
+				if([fTypes.list, fTypes.bool, fTypes.exList, fTypes.selList].indexOf(fType) > -1){
```
Comment (jsmccrumb) [inline js/structured-filter.js]: Alternatively could be (here and 177):
````javascript
if(fType==fTypes.list || fType==fTypes.bool || fType==fTypes.exList || fType==fTypes.selList){}
````
though I think array is more succinct and easier to maintain?

File: js/structured-filter.js [hunk 2]
```diff
@@ -339,6 +341,8 @@ $.widget( 'evol.structFilter', {
 					h+=EvoUI.inputHidden('operator',evoAPI.sInList);
 					this._operator=evoAPI.sInList;
 					break;
+				case fTypes.exList:
+				case fTypes.selList:
```
Comment (jsmccrumb) [inline js/structured-filter.js]: exclusive list and select list follow the same operator rule as boolean (equal)

**Comment by evoluteur** [comment]
Thanks. I like the feature idea. It not fully working yet. I'm looking at it.

**Comment by jsmccrumb** [comment]
Excellent, thanks! If there is anything in particular I can do to help just let me know.

**Pull request merged**